t2.lic

```diff
@@ -41,12 +46,17 @@ class T2
         end
 
         # At this point we know that we need to train the skill
-        echo "***STATUS*** Starting #{trainable['skill']}"
+        if trainable['name'].nil?
+          echo "***STATUS*** Starting (#{trainable['start']}) #{trainable['skill']}"
+        else
+          echo "***STATUS*** Starting (#{trainable['start']}) #{trainable['name']}"
+        end
         execute_actions(trainable['scripts'])
-        update_cooldown trainable['skill'] if has_cooldown? trainable['skill']
+        update_cooldown(trainable['skill']) if has_cooldown?(trainable['skill'])
         break
       end
     end
+    execute_actions(@after_shutdown) unless @after_shutdown.nil?
```

Comment:
If I'm reading this PR right, `@after_shutdown` should never be `nil`, only empty.

Should this check be removed, changed to `.empty?`, or left as-is? Does `execute_actions` handle an empty array correctly?

Reply:
Q on the the preferred style preferences:

```ruby
    if !@after_shutdown.empty?
      echo '***STATUS*** Controlled shutdown, executing shutdown actions.'
      execute_actions(@after_shutdown) 
    end
```
vs
```ruby
    unless @after_shutdown.empty?
      echo '***STATUS*** Controlled shutdown, executing shutdown actions.'
      execute_actions(@after_shutdown) 
    end
```

I tend to avoid `unless`, unless it's at the end, as it feels awkward and confusing when used before the code block itself to me, but that could because I'm not used to it as a Boolean check, but am open to what is considered more standard ruby practices.